tools/tools.go

```diff
@@ -0,0 +1,35 @@
+//go:build tools
+
+// This package contains imports to various tools used (eg, via //go:generate) within this repo,
+// allowing them to be versioned and ensuring their dependncies match what the shim use (specifically
+// for auto-generated protobuf code).
+// Having tools vendored allows `go run <cmd/import/path>` to work without needing to specify a version
+// or download code.
+//
+// Using a dedicate package prevents callers who import github.com/Microsoft/hcsshim from including these
+// in their dependencies.
+//
+// Based on golang [guidance].
+//
+// [guidance]: https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module
```

Comment:
I think maybe we should break this out into a README and include more information about when to add new dependencies in here and how. Thoughts?